### PR TITLE
fix(web): macOS 11 agent string parsing

### LINKED ISSUE #4496: The Keyman script does not work on Mac in Chrome and Opera browsers
The Keyman script at https://keymanweb.com/ does not work on Mac in Chrome and Opera browsers, but works correctly on  Mac in Safari and Firefox browsers.

In Chrome and Opera browsers on Mac these errors is being outputted at https://keymanweb.com/ page:

Uncaught TypeError: Cannot read property 'length' of null
    at g.detect (kmwdevice.ts:92)
    at e.initDevices (kmwutils.ts:72)
    at new e (kmwutils.ts:51)
    at new k (kmwbase.ts:116)
    at kmwbase.ts:639
    at kmwbase.ts:631
kmwuitoolbar.ts:9 Uncaught TypeError: Cannot read property 'ui' of undefined
    at kmwuitoolbar.ts:9
    at kmwuitoolbar.ts:1190
kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at kmwlive.663f7399d98a234f17ddc68d22c8b805.js:386
(index):96 Uncaught ReferenceError: keyman is not defined
    at loadKeyboardFromHash ((index):96)
    at (index):127
helpers.ts:111 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at HTMLDocument.<anonymous> (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:184)
    at j (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at Object.fireWith [as resolveWith] (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at Function.ready (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at HTMLDocument.J (jquery1-11-1.min.4dc834d16a0d219d5c2b8a5b814569e4.js:2)
    at HTMLDocument.sentryWrapped (helpers.ts:87)
2kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at updateSize (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:127)
kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11 Uncaught ReferenceError: keyman is not defined
    at getKeymanWeb (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:11)
    at updateSize (kmwlive.663f7399d98a234f17ddc68d22c8b805.js:127)

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #4496.

Was caused by a suboptimal regex expression.

### CODE REVIEW

File: web/source/kmwdevice.ts [hunk 1]
```diff
@@ -84,11 +84,14 @@ namespace com.keyman {
...
-          let regex = /Intel Mac OS X (10(?:[_\.]\d+)+)/i;
+          let regex = /Intel Mac OS X (\d+(?:[_\.]\d+)+)/i;
           let results = regex.exec(agent);
           
           // Match result:  a version string with components separated by underscores.
-          if(results.length > 1 && results[1]) {
+          if(!results) {
+            console.warn("KMW could not properly parse the user agent string."
+              + "A suboptimal keyboard layout may result.");
+          } else if(results.length > 1 && results[1]) {
```
Comment (mcdurdin) [inline web/source/kmwdevice.ts:94]: What happens with `!results` -- do we end up with an undefined `this.OS`? Is that optimal?

**Review by mcdurdin — APPROVED** [review]
LGTM. We should cherry-pick to 13.0.

File: web/source/kmwdevice.ts [hunk 2]
```diff
@@ -91,6 +91,7 @@ namespace com.keyman {
           if(!results) {
             console.warn("KMW could not properly parse the user agent string."
               + "A suboptimal keyboard layout may result.");
+            this.OS='MacOSX';
```
Comment (mcdurdin) [inline web/source/kmwdevice.ts:94]: Can't this go in the outer block?

Comment (jahorton) [reply web/source/kmwdevice.ts:94]: Unfortunately, no.  That'd break our iOS device detection whenever a user prefers Safari to show desktop pages, which is the iOS default setting on tablets.

There actually _is_ an override that can be set on the WebView, but we don't currently have that in place.  I'd personally prefer to have both in place, just as a failsafe should one or the other break.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.244-beta](https://keyman.com/downloads/releases/beta/14.0.244)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.19-alpha](https://keyman.com/downloads/releases/alpha/15.0.19)

**Pull request merged**